### PR TITLE
Add Windows-specific API to get device container ID

### PR DESCRIPTION
**Pull request by triplef**
As discussed in https://github.com/libusb/hidapi/discussions/252#discussioncomment-2089707.

### CODE REVIEW

File: windows/hidapi_winapi.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+		/** @brief Get the container ID for a HID device.
+
+			@ingroup API
+			@param dev A device handle returned from hid_open().
+			@param guid The device's container ID on return.
```
Comment (Youw) [inline windows/hidapi_winapi.h:39]:
```suggestion
			@param guid The device's container ID on return.

			@note Make sure to include `<Windows.h>` before including `hidapi_winapi.h`.
```

File: windows/hidapi_winapi.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+			@returns
+				This function returns 0 on success and -1 on error.
+		*/
+		int HID_API_EXPORT_CALL hid_winapi_get_container_id(hid_device *dev, GUID *guid);
```
Comment (Youw) [inline windows/hidapi_winapi.h:44]:
```suggestion
		int HID_API_EXPORT_CALL hid_winapi_get_container_id(hid_device *dev, GUID *container_id);
```

File: windows/hid.c [hunk 3]
```diff
@@ -1048,6 +1048,44 @@ int HID_API_EXPORT_CALL HID_API_CALL hid_get_indexed_string(hid_device *dev, int
...
+	/* Open devnode from device id */
+	cr = CM_Locate_DevNodeW(&dev_node, (DEVINSTID_W)device_id, CM_LOCATE_DEVNODE_NORMAL);
+	if (cr != CR_SUCCESS)
+		goto end;
+
+	/* Get the container id from devnode */
+	container_id = hid_internal_get_devnode_property(dev_node, &DEVPKEY_Device_ContainerId, DEVPROP_TYPE_GUID);
+	if (container_id) {
+		*guid = *container_id;
+	}
```
Comment (Youw) [inline windows/hid.c:1079]: call `CM_Get_DevNode_PropertyW` directly and pass `guid` - you'll avoid extra memory allocation/copy/free

File: windows/hidapi_winapi.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+#include <stdint.h>
+
+#include "hidapi.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+		/** @brief Get the container ID for a HID device.
```
Comment (Youw) [inline windows/hidapi_winapi.h:35]: Add mode description/details. Make a reference to `DEVPKEY_Device_ContainerId`

**Review by Youw — COMMENTED** [review]
Looks good

File: windows/hidapi_winapi.h (hunk 1 above)
Comment (triplef) [reply windows/hidapi_winapi.h:43]: I included `guiddef.h`, which is all that should be needed to include `hidapi_winapi.h`.

**Pull request merged**